#! /bin/sh -e

# DP: change location of capiinit conffile to /etc/isdn/capi.conf
# DP: add firmware information for recent AVM cards

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p1 < $0
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p1 < $0
        ;;
    *)
	echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
        exit 1
esac
exit 0

--- isdnutils-3.3.orig/capiinit/capiinit.c
+++ isdnutils-3.3/capiinit/capiinit.c
@@ -91,7 +91,7 @@
 
 static int patchdebug = 0;	/* -d || -debug */
 static int silent = 0;		/* -s || -silent */
-char *configfilename = "/etc/capi.conf";
+char *configfilename = "/etc/isdn/capi.conf";
 
 /* ---------------- utils -------------------------------------------- */
 
--- ./capiinit/capi.conf~	2001-04-18 12:21:42.000000000 +0200
+++ ./capiinit/capi.conf	2005-01-09 01:18:17.593925695 +0100
@@ -1,13 +1,23 @@
-# card		file	proto	io	irq	mem	cardnr	options
-#b1isa		b1.t4	DSS1	0x150	7	-	-	P2P
-b1pci		b1.t4	DSS1	-	-	-	-
-c4		c4.bin	DSS1	-	-	-	-
-c4		-	DSS1	-	-	-	-
-c4		-	DSS1	-	-	-	-	P2P
-c4		-	DSS1	-	-	-	-	P2P
-#c2		c2.bin	DSS1	-	-	-	-
-#c2		-	DSS1	-	-	-	-
-#t1isa		t1.t4	DSS1	0x340	9	-	0
-#t1pci		t1.t4	DSS1	-	-	-	-
-#fcpci		-	-	-	-	-	-
-#fcclassic	-	-	0x150	10	-	-
+# card		file		proto	io	irq	mem	cardnr	options
+#b1isa		b1.t4		DSS1	0x150	7	-	-	P2P
+#b1pci		b1.t4		DSS1	-	-	-	-
+#c4		c4.bin		DSS1	-	-	-	-
+#c4		-		DSS1	-	-	-	-
+#c4		-		DSS1	-	-	-	-	P2P
+#c4		-		DSS1	-	-	-	-	P2P
+#c2		c2.bin		DSS1	-	-	-	-
+#c2		-		DSS1	-	-	-	-
+#t1isa		t1.t4		DSS1	0x340	9	-	0
+#t1pci		t1.t4		DSS1	-	-	-	-
+#fcpci		-		-	-	-	-	-
+#fcpcmcia	-		-	-	-	-	-
+#fcusb		-		-	-	-	-	-
+#fxusb		-		-	-	-	-	-
+#fcclassic	-		-	0x150	10	-	-
+#fcdsl		fdslbase.bin	DSS1	-	-	-	-
+#fcdsl2		fdsl2base.bin	DSS1	-	-	-	-
+#fcdslsl	fdssbase.bin	DSS1	-	-	-	-
+#fcdslslusb	fdlubase.frm	DSS1	-	-	-	-
+#fcdslusba	fdlabase.frm	DSS1	-	-	-	-
+#fcdslusb	fdsubase.frm	DSS1	-	-	-	-
+#fcdslusb	fdsubase.frm	DSS1	-	-	-	-
